crates/sui-core/src/authority/authority_per_epoch_store.rs

```diff
@@ -2341,27 +2325,22 @@ impl AuthorityPerEpochStore {
             shared_input_next_versions.into_iter(),
         )?;
 
-        let lock_and_final_round =
-            self.process_end_of_publish_transactions(batch, end_of_publish_transactions)?;
+        let lock = self.process_end_of_publish_transactions(batch, end_of_publish_transactions)?;
+        let (lock, final_round) =
+            self.process_end_of_all_transactions(batch, lock, commit_has_deferred_txns)?;
```

Comment:
`process_end_of_publish_transactions` and `process_end_of_all_transactions` are only called here, and they have this somewhat tricky dance they do to pass the lock between them. It seems like they could be simplified if you just combined the two functions into one.

Reply:
done